pkg/asset/cluster/azure/azure.go

```diff
@@ -12,3 +21,44 @@ func Metadata(config *types.InstallConfig) *azure.Metadata {
 		Region: config.Platform.Azure.Region,
 	}
 }
+
+// PreTerraform performs any infrastructure initialization which must
+// happen before Terraform creates the remaining infrastructure.
+func PreTerraform(ctx context.Context, clusterID *installconfig.ClusterID, installConfig *installconfig.InstallConfig) error {
```

Comment:
the PreTerraform is designed for certain actions that need to be done before running terraform, because terraform can't achieve those.

These should be rare.

I think a better location for this validation is actually in `pkg/asset/installconfig/azure/validations.go`

Reply:
I am a bit confused by this. I was under the impression the validation code does not get run when the install runs, but when one does a "go test". Is this not correct?

Reply:
As for adding this to PreTerraform, isn't that ideal here? We want to check that the cluster does not currently existing before generating the terraform.

Reply:
> https://github.com/openshift/installer/pull/3120#discussion_r421010673
```
the PreTerraform is designed for certain actions that need to be done before running terraform, because terraform can't achieve those.
```

```
As for adding this to PreTerraform, isn't that ideal here? We want to check that the cluster does not currently exist before generating the terraform.
```
As I tried to explain before what is allowed functionality of PreTerraform functions, the use of it is wrong based on that.

https://github.com/openshift/installer/blob/master/pkg/asset/installconfig/platformprovisioncheck.go is the correct place to perform these checks.

Reply:
> I am a bit confused by this. I was under the impression the validation code does not get run when the install runs, but when one does a "go test". Is this not correct?

hmm.. I am a little confused by this comment, can you expand a little?
> validation code does not get run when the install runs
validation always runs, there are different places to run different types of validations to allow for various workflows for users...  which one are you specifically talking about?

> but when one does a "go test".
that is unit testing of the code, how is that related to installing?

Reply:
@jhixson74 in general sense, this PR is preventing users from installing with a cluster name when we think there is already a cluster running for that name right?

So do you think.
A. all customers, who are installing clusters using installer and who are only using the installer to create ignition configs and will be installing clusters will benefit from this validation?
=> if yes, then we should keep this validation in `pkg/asset/installconfig/azure/validations.go`

b. only the custmers that are using the installer to install the clusters and NOT the users trying to create ignition configs will benefit from this validation?
=> if yes, then we should keep this validation in `pkg/asset/installconfig/platformprovisioncheck.go`, i.e. call this function in that asset.

i'm happy with either one.

Reply:
I think option B is the best. That is what I've chosen. I have placed the function ValidateDNSRecordsDontExist() in pkg/asset/installconfig/azure/validation.go. This is called inside of pkg/asset/installconfig/platformprovisioncheck.go. I hope this is sufficient. Let me know if this seems kind of dirty. It made sense to me ;-)